app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/HorizontalSnapScrollView.kt

```diff
@@ -0,0 +1,270 @@
+                }
+                //left to right
+                else if (event2.x - event1.x > SWIPE_MIN_DISTANCE && exceedsVelocityThreshold) {
+                    scrollToColumn(activeColumnIndex - columns, false)
+                    return true
+                }
+            } catch (exception: Exception) {
+                MyApp.LogDebug(LOG_TAG, "There was an error processing the Fling event:" + exception.message)
+            }
+
+            return super.onFling(event1, event2, velocityX, velocityY)
+        }
+    }
+
+    override fun onInterceptTouchEvent(event: MotionEvent): Boolean {
+        gestureDetector.onTouchEvent(event)
+        return super.onInterceptTouchEvent(event)
+    }
+
+    /**
+     * Sets the rooms count so it can be used to calculate
+     * the room column dimensions in the next layout phase.
+     */
+    fun setRoomsCount(@IntRange(from = 1) roomsCount: Int) {
+        this.roomsCount = roomsCount
+    }
+
+    fun scrollToColumn(col: Int, fast: Boolean) {
+        val maxColValue = if (columnWidth == 0) 0 else (firstChild.measuredWidth - columnWidth) / columnWidth
+        val col = min(max(col, 0), maxColValue)
+
+        val scrollTo = col * columnWidth
+
+        MyApp.LogDebug(LOG_TAG, "scroll to col " + col + "/" + scrollTo + " " + firstChild.measuredWidth)
+
+        if (fast) {
+            scrollTo(scrollTo, 0)
+            if (roomNames != null) {
+                roomNames!!.scrollTo(scrollTo, 0)
+            }
+        } else {
+            post { smoothScrollTo(scrollTo, 0) }
+        }
+        activeColumnIndex = col
+    }
+
+    private inner class OnTouchListener : View.OnTouchListener {
+        override fun onTouch(view: View, event: MotionEvent) =
+            if (gestureDetector.onTouchEvent(event)) {
+                false
+            } else if (
+                event.action == MotionEvent.ACTION_UP ||
+                event.action == MotionEvent.ACTION_CANCEL
+            ) {
+                val scrollX = event.x.toInt()
+                val distance = scrollX - xStart
+
+                var tempColumnIndex = activeColumnIndex
+
+
+                MyApp.LogDebug(LOG_TAG,
+                    "column width:$columnWidth " +
+                            "scrollX:$scrollX " +
+                            "distance:$distance " +
+                            "active column index:$activeColumnIndex"
+                )
+
+
+                if (displayColumnCount > 1) {
+                    val columnDistance = (abs(distance.toFloat()) / columnWidth).roundToInt()
+
+                    MyApp.LogDebug(LOG_TAG,"column distance: $columnDistance")
+
+                    tempColumnIndex = if (distance > 0) {
+                        activeColumnIndex - columnDistance
+                    } else {
+                        activeColumnIndex + columnDistance
+                    }
+                } else {
+                    if (abs(distance) > columnWidth / 4) {
```

Comment:
Can we please move the magic number `4` into a constant? What does it mean? What does it stand for? The name of the constant should clarify this.

Reply:
Please use `const` and move it to the companion object and capitalize it, e.g. `MIN_PORTRAIT_SCROLLING_PERCENTAGE`. Why not `0.25` instead of `1 / 4`?